### PR TITLE
deploy: remove deprecated generatedeploymentconfig api

### PR DESCRIPTION
**Pull request by mfojtik**
Needs a deprecation notice in https://github.com/openshift/openshift-docs/issues/4906
Need public announcement.
Need a release notes.

According to @kargakis this API is not used at all so we should nuke it.

### CODE REVIEW

**Comment by kargakis** [comment]
Superseded by instantiate ever since 3.4.

**Comment by mfojtik** [comment]
updated bootstrap policy test

**Comment by mfojtik** [comment]
@kargakis all green, ship this?

(after this merge, I will send mail to aos lists and update the openshift-docs issue)

File: pkg/authorization/apis/authorization/deprecated_fields.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ var (
 	groupsToResources = map[string][]string{
 		buildGroupName:       {"builds", "buildconfigs", "buildlogs", "buildconfigs/instantiate", "buildconfigs/instantiatebinary", "builds/log", "builds/clone", "buildconfigs/webhooks"},
 		imageGroupName:       {"imagestreams", "imagestreammappings", "imagestreamtags", "imagestreamimages", "imagestreamimports"},
-		deploymentGroupName:  {"deploymentconfigs", "generatedeploymentconfigs", "deploymentconfigrollbacks", "deploymentconfigs/log", "deploymentconfigs/scale"},
+		deploymentGroupName:  {"deploymentconfigs", "deploymentconfigrollbacks", "deploymentconfigs/log", "deploymentconfigs/scale"},
```
Comment (kargakis) [inline pkg/authorization/apis/authorization/deprecated_fields.go]: We probably want to deprecate `deploymentconfigrollbacks` too in favor of `deploymentconfigs/rollback` but it's a separate issue.

Comment (mfojtik) [inline pkg/authorization/apis/authorization/deprecated_fields.go]: +1 i saw it in master config but want to track that as separate thing

Comment (kargakis) [reply pkg/authorization/apis/authorization/deprecated_fields.go]: Is this tracked anywhere?

**Pull request merged**